src/OutputReaders/extract_field_time_series.jl

```diff
@@ -24,15 +25,32 @@ end
 extract_field_time_series(f::FieldTimeSeries) = f
 
 # For types that do not contain `FieldTimeSeries`, halt the recursion
-CannotPossiblyContainFTS = (:Number, :AbstractArray)
+CannotPossiblyContainFTS = (:Number, :AbstractArray, :AbstractGrid, :AbstractField)
 
 for T in CannotPossiblyContainFTS
     @eval extract_field_time_series(::$T) = nothing
 end
 
 # Special recursion rules for `Tuple` and `Field` types
-extract_field_time_series(t::AbstractField)     = Tuple(extract_field_time_series(getproperty(t, p)) for p in propertynames(t))
 extract_field_time_series(t::AbstractOperation) = Tuple(extract_field_time_series(getproperty(t, p)) for p in propertynames(t))
-
 extract_field_time_series(t::Union{Tuple, NamedTuple}) = map(extract_field_time_series, t)
 
+# Special extract for Fields with FTSBC
+const WFTSBCS = FieldBoundaryConditions{<:FTSBC}
+const EFTSBCS = FieldBoundaryConditions{<:Any, <:FTSBC}
+const SFTSBCS = FieldBoundaryConditions{<:Any, <:Any, <:FTSBC}
+const NFTSBCS = FieldBoundaryConditions{<:Any, <:Any, <:Any, <:FTSBC}
+const BFTSBCS = FieldBoundaryConditions{<:Any, <:Any, <:Any, <:Any, <:FTSBC}
+const TFTSBCS = FieldBoundaryConditions{<:Any, <:Any, <:Any, <:Any, <:Any, <:FTSBC}
+const IFTSBCS = FieldBoundaryConditions{<:Any, <:Any, <:Any, <:Any, <:Any, <:Any, <:FTSBC}
+
+const FieldBCsFTS = Union{WFTSBCS, EFTSBCS, SFTSBCS, NFTSBCS, BFTSBCS, TFTSBCS, IFTSBCS}
+const FieldFTS = Field{LX, LY, LZ, O, G, I, D, T, <:FieldBCsFTS} where {LX, LY, LZ, O, G, I, D, T}
```

Comment:
I don't think this is necessary if you fix the issue with `flattened_unique_values`, is it? You merely need `extract_field_time_series` for `BoundaryCondition` and that's it.

Reply:
I am suspecting that is the issue but haven't confirmed it yet. For sure, this removes the allocations in toto and I think we want something like this anyways because it is a little more explicit.

Reply:
It makes things hard to change though

Reply:
So it has a downside for maintainence. We should avoid complex hardcoded patterns like this if possible so that the codebase remains as nimble as possible, but of course if we can't eliminate allocations while also supporting FTS boundary conditions then we have to accept its downsides.

Reply:
Or rather, we have to see this as a necessary complexity of FTS boundary conditions and then reason about whether the complexity is worth supporting FTS boundary conditions